rpc-server/src/metrics.rs

```diff
@@ -121,6 +121,56 @@ lazy_static! {
 
 }
 
+/// Help method to increment block category metrics
+/// Main idea is to have a single place to increment metrics
+/// It should help to analyze the most popular requests
+/// And build s better caching strategy
+pub async fn increase_block_category_metrics(
+    data: &jsonrpc_v2::Data<crate::config::ServerContext>,
+    block_reference: &near_primitives::types::BlockReference,
+    block_height: Option<u64>,
+) {
+    match block_reference {
+        near_primitives::types::BlockReference::BlockId(_) => {
+            let final_block = data.blocks_info_by_finality.final_cache_block().await;
+            let expected_earliest_available_block =
+                final_block.block_height - 5 * data.genesis_info.genesis_config.epoch_length;
+            if block_height.unwrap_or_default() > expected_earliest_available_block {
+                // This is request to regular nodes which includes 5 last epochs
+                REQUESTS_COUNTER.with_label_values(&["regula_block"]).inc();
+            } else {
+                // This is request to archive nodes which includes oldest blocks than 5 last epochs
+                REQUESTS_COUNTER.with_label_values(&["archive_block"]).inc();
+            }
+        }
+        near_primitives::types::BlockReference::Finality(finality) => {
+            match finality {
+                // Increase the FINAL_REQUESTS_TOTAL metric
+                // if the request has final finality
+                near_primitives::types::Finality::DoomSlug
+                | near_primitives::types::Finality::Final => {
+                    REQUESTS_COUNTER.with_label_values(&["final"]).inc();
+                }
+                // Increase the OPTIMISTIC_REQUESTS_TOTAL metric
```

Comment:
This comment seems misleading, it mentions `OPTIMISTIC_REQUESTS_TOTAL` but increases the `REQUESTS_COUNTER`